### DISCUSSION TITLE
How to use HTTP Auth JWT

### DISCUSSION THREAD

**Discussion by christianmahardhika** [gofr-dev/gofr#785] [Q&A] [answered]
I’m working on a GoFr-based web application and need to implement JWT (JSON Web Token) authentication for securing my HTTP endpoints. Could you provide guidance on how to set up JWT authentication in GoFr? Specifically, I’m looking for a step-by-step approach to generate, validate, and use JWTs for protecting my API routes.

Would you like more details on how to implement this in GoFr?

**Comment by srijan-27** [comment, marked as answer]
@christianmahardhika - Did you take a look at the documentation: https://gofr.dev/docs/advanced-guide/http-authentication,
It would be helpful.

**Reply by christianmahardhika** [reply]
I've read the documentation carefully but unfortunately, I not found http-authentication that relate to JWT (from header Bearer Token). I am afraid I just found Basic Auth and Api Key.

**Reply by srijan-27** [reply]
I think you missed the 3rd heading of the page.
The page consists of 3 parts. First is Basic Auth, second is API Key, and the third is `OAuth` which you can refer for `Bearer token` auth.
